### PR TITLE
Testing API

### PR DESCRIPTION
**Pull request by andrewstanfordjason**
api for testing dsp and doing development

### CODE REVIEW

File: lib_dsp/api/dsp_testing.h [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+/*
+ * Random number generation
+ */
+int16_t  dsp_rand_int16(unsigned *r);
+uint16_t dsp_rand_uint16(unsigned *r);
+int32_t  dsp_rand_int32(unsigned *r);
+uint32_t dsp_rand_uint32(unsigned *r);
+int64_t  dsp_rand_int64(unsigned *r);
+uint64_t dsp_rand_uint64(unsigned *r);
+//TODO complex
```
Comment (oscarbailey-xmos) [inline lib_dsp/api/dsp_testing.h]: Could you make an issue for this TODO if it won't be included in this PR?

File: lib_dsp/src/testing/dsp_testing_rand.xc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#include <xclib.h>
+
+/*
+ * Random number generation
+ */
+
+#define CRC_POLY (0xEB31D82E)
+
+int16_t  dsp_rand_int16(unsigned *r){
+    crc32(*r, -1, CRC_POLY);
```
Comment (oscarbailey-xmos) [inline lib_dsp/src/testing/dsp_testing_rand.xc]: https://github.com/xmos/sw_xvf3510/issues/362

Comment (andrewstanfordjason) [reply lib_dsp/src/testing/dsp_testing_rand.xc]: i'll rename it dsp_pseudo_rand...

Comment (andrewstanfordjason) [reply lib_dsp/src/testing/dsp_testing_rand.xc]: Pushed a fix

**Comment by johned0** [comment]
Should we have a test for the test functions, in lib_dsp/tests ?

Should dsp_testing_rand.xc be called dsp_testing_pseudo_rand.xc ?

**Pull request merged**